src/Microsoft.Azure.WebJobs.Extension.Storage/Blobs/Triggers/BlobTriggerAttributeBindingProvider.cs

```diff
@@ -12,58 +12,63 @@
 using Microsoft.Azure.WebJobs.Host.Indexers;
 using Microsoft.Azure.WebJobs.Host.Listeners;
 using Microsoft.Azure.WebJobs.Host.Queues;
-using Microsoft.Azure.WebJobs.Host.Storage;
-using Microsoft.Azure.WebJobs.Host.Storage.Blob;
+using Microsoft.Azure.WebJobs.Host.Queues.Listeners;
 using Microsoft.Azure.WebJobs.Host.Timers;
 using Microsoft.Azure.WebJobs.Host.Triggers;
 using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.WindowsAzure.Storage.Blob;
 
 namespace Microsoft.Azure.WebJobs.Host.Blobs.Triggers
 {
     internal class BlobTriggerExtensionConfig : IExtensionConfigProvider
     {
-        private IStorageAccountProvider _accountProvider;
+        private XStorageAccountProvider _accountProvider;
+        private BlobTriggerAttributeBindingProvider _triggerBinder;
 
-        public void Initialize(ExtensionConfigContext context)
+        public BlobTriggerExtensionConfig(XStorageAccountProvider accountProvider, BlobTriggerAttributeBindingProvider triggerBinder)
         {
-            _accountProvider = context.Config.GetService<IStorageAccountProvider>();
+            _accountProvider = accountProvider;
+            _triggerBinder = triggerBinder;
+        }
 
+        public void Initialize(ExtensionConfigContext context)
+        {
             var rule = context.AddBindingRule<BlobTriggerAttribute>();
-            rule.BindToTrigger<IStorageBlob>();
+            rule.BindToTrigger<ICloudBlob>(_triggerBinder);
 
-            rule.AddConverter<IStorageBlob, DirectInvokeString>(blob => new DirectInvokeString(blob.GetBlobPath()));
-            rule.AddConverter<DirectInvokeString, IStorageBlob>(ConvertFromInvokeString);
+            rule.AddConverter<ICloudBlob, DirectInvokeString>(blob => new DirectInvokeString(blob.GetBlobPath()));
+            rule.AddConverter<DirectInvokeString, ICloudBlob>(ConvertFromInvokeString);
 
             // Common converters shared between [Blob] and [BlobTrigger]
 
             // Trigger already has the IStorageBlob. Whereas BindToInput defines: Attr-->Stream. 
             //  Converter manager already has Stream-->Byte[],String,TextReader
-            context.AddConverter<IStorageBlob, Stream>(ConvertToStreamAsync);
+            context.AddConverter<ICloudBlob, Stream>(ConvertToStreamAsync);
 
-            // Blob type is a property of an existing blob. 
-            context.AddConverter(new StorageBlobToCloudBlobConverter());
-            context.AddConverter(new StorageBlobToCloudBlockBlobConverter());
-            context.AddConverter(new StorageBlobToCloudPageBlobConverter());
-            context.AddConverter(new StorageBlobToCloudAppendBlobConverter());
+            // Blob type is a property of an existing blob.             
+            // $$$ did we lose CloudBlob. That's a base class for Cloud*Blob, but does not implement ICloudBlob? 
```

Comment:
Yes - CloudBlob doesn't implement ICloudBlob. We need to retain the ability to bind to CloudBlob base type